### PR TITLE
feat: add image digest value for datadog helm chart

### LINKED ISSUE #714: Image digest not supported - Datadog helm chart
**Describe what happened:**
Image Digest not supported.

**Describe what you expected:**
Need a way to use an image digest for a given image.
If I pass an image digest as a tag, it will render in a way that does not work as intended.
For example:
`"{{.Values.agents.image.repository}}:{{.Values.agents.image.tag}}"`
Would render something like this:
`myimagerepository:sha256:4a1c4b21597c1b4415bdbecb28a3296c6b5e23ca4f9feeb599860a1dac6a0108`

This will not work as expected, the result I want is something as follows:
`myimagerepository@sha256:4a1c4b21597c1b4415bdbecb28a3296c6b5e23ca4f9feeb599860a1dac6a0108`

**Steps to reproduce the issue:**

Pass an image digest as a value under **value.yaml** for any of the image.tag fields supported and do a `helm template`.

### PR DESCRIPTION
**Pull request by francRang**
#### What this PR does / why we need it:
PR will add support for image digest fields when passing it instead of of image tag. Required for when adding image signatures.
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #
fixes #714
#### Special notes for your reviewer:
This PR does not add the digest field for the kube-state-metrics values under values.yaml as this depends on the kube-state-metrics chart, which does not support image digest at this moment. A separate PR will be created to add this field once the kube-state-metrics chart supports such field.
#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] Chart Version bumped
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

**Review by hestonhoffman — APPROVED** [review]

File: charts/datadog/values.yaml
Comment (clamoriniere) [inline charts/datadog/values.yaml:700]: It can be good to say that it takes precedence on the `tag`.
can be added also to other images `digest`

**Review by clamoriniere — COMMENTED** [review]
Hi @francRang 
Thanks for opening this pull request.

The change looks good, juste 2 comments:
* add the information that digest take precedence on `tag`.
* add a CI test in the folder https://github.com/DataDog/helm-charts/tree/main/charts/datadog/ci.

🙇

File: charts/datadog/values.yaml
Comment (francRang) [reply charts/datadog/values.yaml]: Added requested changes

**Comment by francRang** [comment]
Hi @clamoriniere applied the suggested changes, let me know if the PR is good to go

**Review by clamoriniere — APPROVED** [review]
@francRang the changes look good 👍

**Comment by francRang** [comment]
I cannot merge it myself 😞
It says I am not authorized.
@clamoriniere

**Comment by clamoriniere** [comment]
> I cannot merge it myself 😞 It says I am not authorized. @clamoriniere

I was waiting all CI tests to pass before merging it. All good now.

Thanks again for this contribution

**Pull request merged**
